### PR TITLE
Check if user with 'User Administrator' role can delete group.

### PR DESCRIPTION
**Pull request by ndehadrai**
This Automation checks if test user with 'User Administrator'
role can delete group and no errors to be returned.

Fixes: https://pagure.io/freeipa/issue/6884

Signed-off-by: Nikhil Dehadrai <ndehadra@redhat.com>

### CODE REVIEW

**Comment by tiran** [comment]
pylint issues:
```
************* Module ipatests.test_integration.test_user_permissions
ipatests/test_integration/test_user_permissions.py:104: [E1003(bad-super-call), TestGroupDelUserAdmin.install] Bad first argument 'TestUserPermissions' given to super())
ipatests/test_integration/test_user_permissions.py:122: [W0612(unused-variable), TestGroupDelUserAdmin.install] Unused variable 'testadmin_kinit')
```

Please remember to always run ``make fastcheck`` before you submit a PR. This will catch almost all pylint and pycodecheck issues locally.

**Comment by tiran** [comment]
There is one pylint error left:

```
************* Module ipatests.test_integration.test_user_permissions
ipatests/test_integration/test_user_permissions.py:120: [E1101(no-member), TestUserPermissions.test_delete_group_by_user_administrator] Instance of 'TestUserPermissions' has no 'testadmin' member)
```

File: ipatests/test_integration/test_user_permissions.py [hunk 1]
```diff
@@ -93,3 +93,36 @@ def test_stageuser_show_as_alternate_admin(self):
...
+
+        # kinit as testuser
+        self.master.run_command(['kinit', testadmin],
+                                stdin_text=testadmin_password_confirmation)
+
+        # Create a test group
+        self.master.run_command(['ipa', 'group-add', testgroup])
+
+        # Call ipa-group-del to check if user can delete group
+        result = self.master.run_command(['ipa', 'group-del', testgroup])
```
Comment (flo-renaud) [inline ipatests/test_integration/test_user_permissions.py]: Just a nitpick, but I would also add a check for result.returncode == 0

Comment (Rezney) [reply ipatests/test_integration/test_user_permissions.py]: @flo-renaud that should not be needed as there is "raiseonerr=True" by default.

https://github.com/encukou/pytest-multihost/blob/862f24d536ffde32cc636151e833c38d909584b2/pytest_multihost/host.py#L202

But as you told that is just a nitpick. @ndehadrai  please leave it as it is...

Comment (flo-renaud) [reply ipatests/test_integration/test_user_permissions.py]: @Rezney of course, you are obviously right!

File: ipatests/test_integration/test_user_permissions.py [hunk 2]
```diff
@@ -149,3 +149,38 @@ def test_installclient_as_user_admin(self):
...
+        assert 'Deleted group' in result.stdout_text
+        if result.returncode == 0:
+            print("Group-del is successful with 'user administrator' profile")
```
Comment (rcritten) [inline ipatests/test_integration/test_user_permissions.py]: No need to print anything

ipatests/test_integration/test_user_permissions.py:186: [E1601(print-statement), TestInstallClientNoAdmin.test_delete_group_by_user_administrator] print statement used)

Asserting result.returncode == 0 should be sufficient. If a user attempts to delete a group and they lack permission a non-zero returncode should be set.

**Comment by rcritten** [comment]
@ndehadrai this needs a rebase.

**Comment by mrizwan93** [comment]
@ndehadrai Are you still working on it?

**Comment by flo-renaud** [comment]
Closing in favor of PR #4702

**Pull request closed without merging**